dlp/snippets/deid.py

```diff
@@ -1023,6 +1023,122 @@ def deidentify_with_exception_list(
 # [END dlp_deidentify_exception_list]
 
 
+# [START dlp_deidentify_table_infotypes]
+def deidentify_table_replace_with_info_types(
+    project,
+    table_data,
+    info_types,
+    deid_content_list
+):
+    """ Uses the Data Loss Prevention API to de-identify sensitive data in a
+      table by replacing them with info type.
+
+    Args:
+        project: The Google Cloud project id to use as a parent resource.
+        table_data: Json string representing table data.
+        info_types: A list of strings representing info types to look for.
+            A full list of info type categories can be fetched from the API.
+        deid_content_list: A list of fields in table to de-identify
+
+    Returns:
+        De-identified table is returned;
+        the response from the API is also printed to the terminal.
+
+    Example:
+    table_data = {
+        "header":[
+            "name"
+            "email",
+            "phone number"
+        ],
+        "rows":[
+            [
+                "Robert Frost",
+                "robertfrost@xyz.com",
+                "4232342345"
+            ],
+            [
+                "John Doe",
+                "johndoe@pqr.com",
+                "4253458383"
+            ]
+        ]
+    }
+
+    >> $ python deid.py table_replace_with_infotype \
+        '{"header": ["name", "email", "phone number"],
+        "rows": [["Robert Frost", "robertfrost@xyz.com", "4232342345"],
+        ["John Doe", "johndoe@pqr.com", "4253458383"]]}' \
+        ["PERSON_NAME"] ["name"]
+        >> '{"header": ["name", "email", "phone number"],
+            "rows": [["[PERSON_NAME]", "robertfrost@xyz.com", "4232342345"],
+            ["[PERSON_NAME]", "johndoe@pqr.com", "4253458383"]]}'
```

Comment:
Could you please fix the indentation?
Also why do we list the input table twice? Could we simply write the python call in a way that takes the table_data in a pretty-printed format and not declare the table_data at all?

Reply:
updated; Thank you for reviewing